src/app/loadout-builder/filter/ModPicker.tsx

```diff
@@ -144,165 +145,153 @@ function mapStateToProps() {
   });
 }
 
-interface State {
-  query: string;
-  height?: number;
-  lockedArmor2Mods: LockedArmor2ModMap;
-}
-
 /**
  * A sheet that allows picking a perk.
  */
-class ModPicker extends React.Component<Props, State> {
-  state: State = {
-    query: '',
-    lockedArmor2Mods: copy(this.props.lockedArmor2Mods),
-  };
-  private itemContainer = React.createRef<HTMLDivElement>();
-  private filterInput = React.createRef<SearchFilterRef>();
-
-  componentDidMount() {
-    if (this.itemContainer.current) {
-      this.setState({ height: this.itemContainer.current.clientHeight });
-    }
-    if (!this.props.isPhonePortrait && this.filterInput.current) {
-      this.filterInput.current.focusFilterInput();
+function ModPicker({
+  defs,
+  mods,
+  language,
+  onClose,
+  isPhonePortrait,
+  lockedArmor2Mods,
+  lbDispatch,
+}: Props) {
+  const [query, setQuery] = useState('');
+  const [lockedArmor2ModsInternal, setLockedArmor2ModsInternal] = useState(copy(lockedArmor2Mods));
+  const filterInput = useRef<SearchFilterRef | null>(null);
+
+  useEffect(() => {
+    if (!isPhonePortrait && filterInput.current) {
+      filterInput.current.focusFilterInput();
     }
-  }
+  }, [isPhonePortrait, filterInput]);
+
+  const onModSelected = useCallback(
```

Comment:
I wrapped anything going into the ModPickerSection in either a useMemo or useCallback. These were private functions before.